### PR TITLE
Ability to deploy shortcode templates

### LINKED ISSUE #7270: Shortcode Templates Feature needs deployment plans and recipes
Appears to have been forgotten todo

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #7270

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Shortcodes/Startup.cs [hunk 1]
```diff
@@ -68,6 +70,10 @@ public ShortcodeTemplatesStartup(IOptions<AdminOptions> adminOptions)
 
         public override void ConfigureServices(IServiceCollection services)
         {
+            services.AddTransient<IDeploymentSource, AllShortcodeTemplatesDeploymentSource>();
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Shortcodes/Startup.cs:73]: Should be registered in a different startup, using `RequiresFeatures("OrchardCore.Deployment", "OrchardCore.Shortcodes.Templates")`

File: src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Items/AllShortcodeTemplatesDeploymentStep.Thumbnail.cshtml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+@model dynamic
+
+<h4 class="card-title">@T["All Shortcode Templates"]</h4>
+<p>@T["Exports every shortcode template."]</p>
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Items/AllShortcodeTemplatesDeploymentStep.Thumbnail.cshtml:4]:
```suggestion
<p>@T["Exports all shortcode templates."]</p>
```

**Review by deanmarcussen — COMMENTED** [review]
Please try your prs before wasting time asking for a review.

The recipe step is not registered. If you tried it you would know this

**Comment by hishamco** [comment]
> Please try your prs before wasting time asking for a review.

Dean I don't have a shortcode template at time of creating the PR but I added a deployment target and when the card for export shows up I thought it should, I will try to create a shortcode template because I didn't try it before and update the PR

**Comment by hishamco** [comment]
@deanmarcussen I'm unable to create a shortcode template, there's an issue in the categories `vue-multiselect` it doesn't show up and console log shows a bunch of errors for CodeMirror

/cc @agriffard

File: src/OrchardCore.Modules/OrchardCore.Shortcodes/Startup.cs [hunk 3]
```diff
@@ -124,4 +128,18 @@ public override void ConfigureServices(IServiceCollection services)
             });
         }
     }
+
+    [RequireFeatures("OrchardCore.Deployment", "OrchardCore.Shortcodes.Templates")]
+    public class ShortcodeTemplatesDeployementStartup : StartupBase
+    {
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            services.AddRecipeExecutionStep<ShortcodeTemplateStep>();
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Shortcodes/Startup.cs:137]: This line goes in the normal startup, as Recipes are their own module (not part of Deployment) and recipes are AlwaysOn

**Comment by hishamco** [comment]
What about the above issue Dean? Is it a regression bug?

**Comment by hishamco** [comment]
@agriffard why the Multifield doesn't show up?

File: src/OrchardCore.Modules/OrchardCore.Shortcodes/Startup.cs (hunk 3 above)
Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Shortcodes/Startup.cs:137]: I think it should go to `ShortcodeTemplatesStartup` while it requires `ShortcodeTemplateManager`

Comment (deanmarcussen) [reply src/OrchardCore.Modules/OrchardCore.Shortcodes/Startup.cs:137]: That's what I said

**Comment by deanmarcussen** [comment]
Try merging dev if your cdn is still not working @hishamco 

If not, just disable the cdn.

Not sure what this has to do with the Multifield? what is that?

**Comment by hishamco** [comment]
> If not, just disable the cdn.

I will try ...

> Not sure what this has to do with the Multifield? what is that?

I mean the categories field for the shortcode template

**Comment by hishamco** [comment]
Finally the shortcode categories filed show up after merge dev

**Comment by hishamco** [comment]
Tested and everything is looks good

**Comment by hishamco** [comment]
@deanmarcussen your final approval please

**Comment by davidhayden** [comment]
@deanmarcussen @hishamco @agriffard Great to see this being worked on as I just noticed deployment steps for shortcode templates were missing this morning. Thanks for all your efforts! 👍

**Review by deanmarcussen — APPROVED** [review]

**Pull request merged**